packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -228,34 +229,67 @@ export async function handler(event: SQSEvent) {
         log(
           `Calling converter on url ${converterUrl} with params ${JSON.stringify(converterParams)}`
         );
-        const convertPayloadToFHIR = () =>
-          executeWithNetworkRetries(
-            () =>
-              fhirConverter.post(converterUrl, payloadClean, {
-                params: converterParams,
-                headers: { "Content-Type": TXT_MIME_TYPE },
-              }),
-            {
-              // No retries on timeout b/c we want to re-enqueue instead of trying within the same lambda run,
-              // it could lead to timing out the lambda execution.
-              log,
+
+        const partitionedPayload = partitionPayload(payloadClean);
+        const convertPayloadToFHIR = async () => {
+          const combinedBundle: FHIRBundle = {
+            resourceType: "Bundle",
+            type: "batch",
+            entry: [],
+          };
+
+          if (partitionedPayload.length > 1) {
+            log(`The file was partitioned into ${partitionedPayload.length} parts...`);
+          }
+
+          for (let index = 0; index < partitionedPayload.length; index++) {
+            const payload = partitionedPayload[index];
+
+            const res = await executeWithNetworkRetries(
+              () =>
+                fhirConverter.post(converterUrl, payload, {
+                  params: converterParams,
+                  headers: { "Content-Type": TXT_MIME_TYPE },
+                }),
+              {
+                // No retries on timeout b/c we want to re-enqueue instead of trying within the same lambda run,
+                // it could lead to timing out the lambda execution.
+                log,
+              }
+            );
+
+            const conversionResult = res.data.fhirResource;
+
+            if (conversionResult?.entry?.length > 0) {
+              log(
+                `Current partial bundle with index ${index} contains: ${conversionResult.entry.length} resources...`
+              );
+              combinedBundle.entry.push(...conversionResult.entry);
             }
-          );
+          }
+
+          log(`Combined bundle contains: ${combinedBundle.entry.length} resources`);
+          return combinedBundle;
+        };
+
         // The actual payload we send to the Converter
-        const storePayloadInS3 = () =>
-          storePreConversionPayloadInS3({
-            payload: payloadClean,
-            preConversionFilename,
-            message,
-            lambdaParams,
-            log,
+        const storePayloadInS3 = () => {
+          partitionedPayload.forEach((payload, index) => {
+            storePreConversionPayloadInS3({
+              payload,
+              preConversionFilename:
+                index > 0
+                  ? `${preConversionFilename}_part_${index + 1}.xml`
+                  : preConversionFilename,
```

Comment:
Why no `_part_0` for the first item?

Nit: I suggest we stick with 0 index based naming- to match the code.

Also, can we move this to a function, please? Its easier to refer to, read, and test.

Reply:
My thinking was that we don't want to append `part_0`, since most documents will only have just the one part. But now that im thinking about it more, i think we can do that only for documents that consist of more than one part

Reply:
sounds good! just keep in mind, for smaller files, we're gonna have a the file prior to partitioning being the same as `part_00`